conans/client/tools/win.py

```diff
@@ -333,7 +338,16 @@ def vcvars_command(settings, arch=None, compiler_version=None, force=False, vcva
     arch_setting = arch_setting or 'x86_64'
     arch_build = settings.get_safe("arch_build") or detected_architecture()
     if arch_build == 'x86_64':
-        vcvars_arch = {'x86': 'amd64_x86' if int(compiler_version) >= 12 else 'x86',
+        # Only uses x64 tooling if arch_build explicitly defines it, otherwise
+        # Keep the VS default, which is x86 toolset
+        # This will probably be changed in conan 2.0
+        if ((settings.get_safe("arch_build") or
+                os.getenv("PreferredToolArchitecture") == "x64")
```

Comment:
`PreferredToolArchitecture` is not an env var normally, right? in that case, I would name it `CONAN_PREFERRED_TOOL_ARCHITECTURE`. And it has to be documented (and mentioned in the changelog)

Reply:
ok then. I can't find it either, just people saying it works but no-official docs. Anyway this is ok.